cmd/autoscaler/main.go

```diff
@@ -225,15 +225,18 @@ func uniScalerFactoryFunc(endpointsInformer corev1informers.EndpointsInformer,
 func statsScraperFactoryFunc(endpointsLister corev1listers.EndpointsLister,
 	podLister corev1listers.PodLister) asmetrics.StatsScraperFactory {
 	return func(metric *av1alpha1.Metric, logger *zap.SugaredLogger) (asmetrics.StatsScraper, error) {
-		if metric.Spec.ScrapeTarget != "" {
-			podCounter := resources.NewScopedEndpointsCounter(
-				endpointsLister, metric.Namespace, metric.Spec.ScrapeTarget)
-			// TODO(vagababov): while metric name == revision name, we should utilize the proper
-			// values from the labels.
-			podAccessor := resources.NewPodAccessor(podLister, metric.Namespace, metric.Name)
-			return asmetrics.NewStatsScraper(metric, podCounter, podAccessor, logger)
+		if metric.Spec.ScrapeTarget == "" {
+			return nil, nil
 		}
-		return nil, nil
+
+		revisionName, ok := metric.Labels[serving.RevisionLabelKey]
+		if !ok || revisionName == "" {
```

Comment:
```suggestion
		revisionName := metric.Labels[serving.RevisionLabelKey]
		if revisionName == "" {
```

Will yield the same semantics here!

Reply:
yeah I was being consistent with https://github.com/knative/serving/blob/74f8cbcea4ff72e085926d5e73e5efce8368e027/cmd/autoscaler/main.go#L205 but.. you're right :)

Reply:
it's fine either way.